packages/istanbul-lib-instrument/src/visitor.js

```diff
@@ -697,9 +697,18 @@ function programVisitor(types, sourceFilePath = 'unknown.js', opts = {}) {
                 .update(JSON.stringify(coverageData))
                 .digest('hex');
             coverageData.hash = hash;
+            const inputSourceMap = coverageData.inputSourceMap;
+            if (inputSourceMap && Object.getPrototypeOf(inputSourceMap) !== Object.prototype) {
+              console.warn("A non-plain object was passed to istanbul-lib-instrument.")
```

Comment:
@bcoe Feel free to suggest edits to the message...

Reply:
We can also print the constructor name in the warning message, so it is easier to identify what creates such input source map.

Reply:
A good idea, thanks!